### PR TITLE
fix(setup-wizard): donation data

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes a Setup wizard issue.

Also updates the useWizardData hook, so it can use multiple wizards. Before, the hook was using the Reader Revenue wizard data only.

### How to test the changes in this Pull Request:

1. On `master`, visit the Setup wizard, "Services" step
2. Enable "Reader Revenue" and set some donation values (e.g. suggested donation amount)
3. Save by navigating to the next setup step
4. Open the Reader Revenue wizard and observe that the inserted values have not been updated
5. Switch to this branch, redo steps 2 & 3, observe the updated values in Reader Revenue wizard

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201745193114476

### CODE REVIEW

File: assets/wizards/readerRevenue/views/donation/index.js [hunk 1]
```diff
@@ -11,13 +11,14 @@ import { MoneyInput } from '../../components/';
 import { Card, Grid, Button, Notice, ToggleControl, Wizard } from '../../../../components/src';
 import { READER_REVENUE_WIZARD_SLUG } from '../../constants';
 
-export const DontationAmounts = ( { data = {} } ) => {
+export const DontationAmounts = () => {
```
Comment (dkoo) [inline assets/wizards/readerRevenue/views/donation/index.js:14]:
```suggestion
export const DonationAmounts = () => {
```

File: assets/wizards/readerRevenue/views/donation/index.js [hunk 2]
```diff
@@ -108,7 +109,7 @@ const Donation = () => {
 					</Card>
 				) }
 				<Card noBorder>
-					<DontationAmounts data={ wizardData.donation_data } />
+					<DontationAmounts />
```
Comment (dkoo) [inline assets/wizards/readerRevenue/views/donation/index.js:112]:
```suggestion
					<DonationAmounts />
```

File: assets/wizards/setup/views/services/ReaderRevenue.js [hunk 3]
```diff
@@ -37,15 +29,9 @@ const ReaderRevenue = ( { configuration, onUpdate, className } ) => {
 				</p>
 			) : (
 				<>
-					<DontationAmounts
-						data={ configuration.donation_data || {} }
-						onChange={ donation_data => onUpdate( { donation_data } ) }
-					/>
+					<DontationAmounts />
```
Comment (dkoo) [inline assets/wizards/setup/views/services/ReaderRevenue.js:32]:
```suggestion
					<DonationAmounts />
```

File: assets/wizards/setup/views/services/ReaderRevenue.js [hunk 4]
```diff
@@ -2,33 +2,25 @@
...
  * WordPress dependencies.
  */
-import apiFetch from '@wordpress/api-fetch';
-import { useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import { DontationAmounts } from '../../../readerRevenue/views/donation';
```
Comment (dkoo) [inline assets/wizards/setup/views/services/ReaderRevenue.js:15]:
```suggestion
import { DonationAmounts } from '../../../readerRevenue/views/donation';
```

**Review by dkoo — CHANGES_REQUESTED** [review]
Ran into a few issues:

1. This isn't really related to this PR, but while we're working with the Services view during setup...if you skip setting up Jetpack in the prior screen, then the Newsletters action card is totally blank:

Screen Shot 2022-02-01 at 11 12 59 AM

Suggestion: Either hide this card if Jetpack isn't connected at this stage, or show a message saying that Jetpack needs to be connected first (Like the `Connect Jetpack in order to configure Mailchimp` message we show in the dashboard wizard page).

2. In the Services page during setup: if you enable some services, fill out some settings, then proceed to the Design step, then go back to Services, the services you previously toggled on are no longer toggled. Expanding them does reveal the settings you previously set are saved, but the "on" state for each service is not saved. Question: is the toggle even necessary for these Services cards? Toggling them on or off doesn't really serve any functional purpose beyond hiding unnecessary settings fields, right? I would ask if we could just omit the toggle controls and show all settings for each service by default. This would remove a step for setting these options during setup (cc @thomasguillot). If we keep the toggle controls, the state should be saved if we go back to the Services page from another step.

3. I was able to replicate this on two test sites in a row. I ran into this JS console error when trying to continue past the Services screen:

```
Object { code: "invalid_json", message: "The response is not a valid JSON response." }
index.js:55:15
```

I had to refresh the page to try again, after which I was able to re-enter my settings and continue, but after finishing setup and going to the Reader Revenue wizard, my settings weren't saved. I had toggled on the "Set exact monthly donation tiers" option and set values for all tiers, but none of these settings were saved. Stripe settings **were** saved, however. Going back to the `admin.php?page=newspack-setup-wizard#/services` page and repeating the steps a second time did not throw the JS console error and did successfully save the settings.

**Comment by thomasguillot** [comment]
> 1. This isn't really related to this PR, but while we're working with the Services view during setup...if you skip setting up Jetpack in the prior screen, then the Newsletters action card is totally blank

Jetpack shouldn't be needed for this. This sounds like a bug to me.

> 2\. In the Services page during setup: if you enable some services, fill out some settings, then proceed to the Design step, then go back to Services, the services you previously toggled on are no longer on. Toggling them on again does reveal the settings you previously set, but the "on" state for each service is not saved. Question: is the toggle even necessary for these Services cards? Toggling them on or off doesn't really serve any functional purpose beyond hiding unnecessary settings fields, right? I would ask if we could just omit the toggle controls and show all settings for each service by default. This would remove a step for setting these options during setup (cc @thomasguillot). If we keep the toggle controls, the state should be saved if we go back to the Services page from another step.

Losing the toggle sounds once again like a bug to me. The idea behind the toggles was that some publishers might not want to enable these services. If you show them a bunch of empty field then they might end up confused with a feeling of being "forced" to use reader revenue, newsletters, etc...

File: assets/wizards/setup/views/services/ReaderRevenue.js (hunk 4 above)
Comment (adekbadek) [reply assets/wizards/setup/views/services/ReaderRevenue.js]: Done in 7c7b657

File: assets/wizards/setup/views/services/ReaderRevenue.js (hunk 3 above)
Comment (adekbadek) [reply assets/wizards/setup/views/services/ReaderRevenue.js]: Done in 7c7b657

File: assets/wizards/readerRevenue/views/donation/index.js (hunk 2 above)
Comment (adekbadek) [reply assets/wizards/readerRevenue/views/donation/index.js]: Done in 7c7b657

File: assets/wizards/readerRevenue/views/donation/index.js (hunk 1 above)
Comment (adekbadek) [reply assets/wizards/readerRevenue/views/donation/index.js]: Done in 7c7b657

**Comment by adekbadek** [comment]
> if you skip setting up Jetpack in the prior screen, then the Newsletters action card is totally blank

There's no Jetpack setup, the "Integrations" screen was removed in #1453 🤔 After disabling Jetpack, I still do see the Newsletters section. After disabling Newspack Newsletters, I see a plugin installation prompt. That said, error and loading states were missing – added in ded9f242d7cc5e84b17d7c20de3557a8f042c495

>  In the Services page during setup: if you enable some services, fill out some settings, then proceed to the Design step, then go back to Services, the services you previously toggled on are no longer on. 

Fixed in 9e22307b79f805a30e179a674979eca5489c8e95. The code before tried to be too smart (trying to infer if the service is "active") and there was a disconnect between what determined the "enabled" state and how the "enabled" state was saved after toggling it in the UI. Now the toggle states just save options values.

Note that this change is technically non-backwards-compatible, but since the Setup wizard is meant to be ran once (after completion it's unavailable in the UI), I don't this this change should be marked as breaking.

> I ran into this JS console error when trying to continue past the Services screen

I could not reproduce that locally, also when using production-built JS files. Could you share more about the circumstances of this error (which services are enabled, which plugins active)? Also, does this happen specifically on this branch, or on `master` too?

**Comment by dkoo** [comment]
> There's no Jetpack setup, the "Integrations" screen was removed in https://github.com/Automattic/newspack-plugin/pull/1453 🤔 After disabling Jetpack, I still do see the Newsletters section. After disabling Newspack Newsletters, I see a plugin installation prompt. That said, error and loading states were missing – added in https://github.com/Automattic/newspack-plugin/commit/ded9f242d7cc5e84b17d7c20de3557a8f042c495. With that in place, are you seeing an error with Jetpack disconnected?

Whoops, my mistake. I must have gotten used to just clicking past the Integrations step. 😆 I'm still seeing the same behavior on a new site with the latest changes, so here's a new issue: #1487

**Review by dkoo — CHANGES_REQUESTED** [review]
> >  In the Services page during setup: if you enable some services, fill out some settings, then proceed to the Design step, then go back to Services, the services you previously toggled on are no longer on. 

> Fixed in 9e22307b79f805a30e179a674979eca5489c8e95. The code before tried to be too smart (trying to infer if the service is "active") and there was a disconnect between what determined the "enabled" state and how the "enabled" state was saved after toggling it in the UI. Now the toggle states just save options values.

> Note that this change is technically non-backwards-compatible, but since the Setup wizard is meant to be ran once (after completion it's unavailable in the UI), I don't think this change should be marked as breaking.

Confirmed—the services stay toggled on/off if you go back to that step, now. I agree about the non-breaking nature of the change since pretty much no one who has already set up their site will be going back to this page.

Still seeing the `invalid_json` error when saving non-default Reader Revenue options for the first time. Details sent via DM.

**Comment by adekbadek** [comment]
> Still seeing the invalid_json error when saving non-default Reader Revenue options for the first time. Details sent via DM.

As discussed, the fatal was happening here:

> PHP Fatal error:  Uncaught Error: Call to a member function get_countries() on null in […]wp-content/plugins/newspack-plugin/includes/configuration_managers/class-woocommerce-configuration-manager.php:57

I still can't reproduce that, but I've added a check to ensure WC's countries handler is available in 9f1ce7fd0b98fb81a553c98ab53ae746f2e7b379

Also, found some API-relates issues with sending is-service-enabled payload (the booleans were sent as `"1"`/`""`) – fixed in 7551ab11c96f8b8ad655e7ac18c342d4df2c2326

**Comment by adekbadek** [comment]
Added another check for WC'c countries handlers availability in 2b548544a73e779414dc4baabb1279fda43a3d3b

**Review by dkoo — APPROVED** [review]
Looks good after the latest changes! 👍

**Pull request merged**